node/src/syncer.rs

```diff
@@ -827,4 +851,33 @@ mod tests {
 
         (syncer, store, handle)
     }
+
+    async fn handle_session_batch(
+        p2p_mock: &mut MockP2pHandle,
+        remaining_headers: &mut Vec<ExtendedHeader>,
+        mut requests: Vec<(u64, u64)>,
+    ) {
+        let first_start = requests[0].0;
+
+        for _ in 0..8 {
+            let (height, amount, respond_to) =
+                p2p_mock.expect_header_request_for_height_cmd().await;
+
+            let pos = requests
+                .iter()
+                .position(|x| *x == (height, amount))
+                .expect("invalid request");
+            requests.remove(pos);
+
+            let start = (height - first_start) as usize;
+            let end = (height - first_start + amount) as usize;
+            respond_to
+                .send(Ok(remaining_headers[start..end].to_vec()))
+                .map_err(|_| format!("headers [{}, {}]", height, height + amount - 1))
+                .unwrap();
+        }
+
+        // Remove already sent batch
+        remaining_headers.drain(..512);
```

Comment:
Is this `512` coming from `MAX_HEADERS_IN_BATCH` in syncer? maybe you could use the const here

Reply:
it the the sum of all amounts. i changed it to be calculated